sdk/servicebus/service-bus/src/util/errors.ts

```diff
@@ -240,3 +243,39 @@ export function throwTypeErrorIfParameterIsEmptyString(
   log.error(`[${connectionId}] %O`, error);
   throw error;
 }
+
+/**
+ * Logs and Throws an error if the given message cannot be settled.
+ * @param receiver Receiver to be used to settle this message
+ * @param operation Settle operation: complete, abandon, defer or deadLetter
+ * @param isRemoteSettled Boolean indicating if the message has been settled at the remote
+ */
+export function throwIfMessageCannotBeSettled(
+  receiver: MessageReceiver | MessageSession | undefined,
+  operation: DispositionType,
+  isRemoteSettled: boolean
+): void {
+  let errorMessage;
+  if (!receiver) {
```

Comment:
Same concern for receiver closed error.
Think it's better treated off as a separate check and error utility.

Reply:
This utility deals with 3 separate cases and throws 3 different errors.
Since all 3 cases deal with the scenario of errors when settling a message, having a single utility is preferable.

At present the 4 ways to settle a message are the only code paths that use `getReceiver` that gives `undefined` when the receiver is closed. These 4 ways are covered by this utility.

In the future, other consumers of the `getReceiver` will be responsible for their own messages.

Reply:
The case being specifically discussed here is when the internally used `getReceiver()` function on the internal `ClientEntityContext` class failed to provide a receiver.

I would agree on having a separate utility to give an error message for this scenario if there were other cases where this function was used. But, since this is only used when settling messages, I don't want to introduce a utility function that is used exactly once in the entire code base.

If we ever get to using this `getReceiver()` function in any other area, we can revisit this decision.